straxen/plugins/position_reconstruction.py

```diff
@@ -1,204 +1,205 @@
-    """Multilayer Perceptron (MLP) neural net for position reconstruction"""
-    provides = "peak_positions_mlp"
-    algorithm = "mlp"
-
-
-@export
-@strax.takes_config(
-    strax.Option('gcn_model',
-                 help='Neural network model.' 
-                      'If CMT, specify as (CMT_model, (gcn_model, ONLINE), True)))'
-                      'Set to None to skip the computation of this plugin.',
-                 default=("CMT_model", ('gcn_model', "ONLINE"), True)
-                )
-)
-class PeakPositionsGCN(PeakPositionsBaseNT):
-    """Graph Convolutional Network (GCN) neural net for position reconstruction"""
-    provides = "peak_positions_gcn"
-    algorithm = "gcn"
-    __version__ = '0.0.1'
-
-
-@export
-@strax.takes_config(
-    strax.Option('cnn_model',
-                 help='Neural network model.' 
-                      'If CMT, specify as (CMT_model, (cnn_model, ONLINE), True)))'
-                      'Set to None to skip the computation of this plugin.',
-                 default=("CMT_model", ('cnn_model', "ONLINE"), True)
-                )
-)
-class PeakPositionsCNN(PeakPositionsBaseNT):
-    """Convolutional Neural Network (CNN) neural net for position reconstruction"""
-    provides = "peak_positions_cnn"
-    algorithm = "cnn"
-    __version__ = '0.0.1'
-
-
-@export
-@strax.takes_config(
-    strax.Option("default_reconstruction_algorithm",
-                 help="default reconstruction algorithm that provides (x,y)",
-                 default="mlp",
-                 )
-)
-class PeakPositionsNT(strax.MergeOnlyPlugin):
-    """
-    Merge the reconstructed algorithms of the different algorithms 
-    into a single one that can be used in Event Basics.
-    
-    Select one of the plugins to provide the 'x' and 'y' to be used 
-    further down the chain. Since we already have the information
-    needed here, there is no need to wait until events to make the
-    decision.
-    
-    Since the computation is trivial as it only combined the three 
-    input plugins, don't save this plugins output.
-    """
-    provides = "peak_positions"
-    depends_on = ("peak_positions_cnn", "peak_positions_mlp", "peak_positions_gcn")
-    save_when = strax.SaveWhen.NEVER
-    __version__ = '0.0.0'
-
-    def infer_dtype(self):
-        dtype = strax.merged_dtype([self.deps[d].dtype_for(d) for d in self.depends_on])
-        dtype += [('x', np.float32, 'Reconstructed S2 X position (cm), uncorrected'),
-                  ('y', np.float32, 'Reconstructed S2 Y position (cm), uncorrected')]
-        return dtype
-
-    def compute(self, peaks):
-        result = {dtype: peaks[dtype] for dtype in peaks.dtype.names}
-        algorithm = self.config['default_reconstruction_algorithm']
-        if not 'x_' + algorithm in peaks.dtype.names:
-            raise ValueError
-        for xy in ('x', 'y'):
-            result[xy] = peaks[f'{xy}_{algorithm}']
-        return result
+"""Position reconstruction for Xenon-nT"""
```

Comment:
Hmm, looks like git got a bit tricked (it appears as if you replaced the entire file), if we need to track where a specific line was set later, this is a bit more tricky because the git-history gets scrambled, would it be possible to fix this?

Reply:
See https://github.com/XENONnT/straxen/pull/396/files#diff-bfd4351fab05e668e9e4f9cf0f7c1318ada3b2c6cf0bd0fd48523511760bb296

Reply:
I am not sure what happened here, but fixed it